pkg/prometheus/promcfg.go

```diff
@@ -68,7 +68,7 @@ type ConfigGenerator struct {
 }
 
 // NewConfigGenerator creates a ConfigGenerator for the provided Prometheus resource.
-func NewConfigGenerator(logger log.Logger, p monitoringv1.PrometheusInterface, endpointSliceSupported bool) (*ConfigGenerator, error) {
+func NewConfigGenerator(logger log.Logger, p monitoringv1.PrometheusInterface, endpointSliceSupported *bool) (*ConfigGenerator, error) {
```

Comment:
why change from a bool value to a bool pointer?

Reply:
Yeah it's not necessary.

Before I thought we needed to check the value of `ServiceDiscoveryRole` inside `NewConfigGenerator()`. If `ServiceDiscoveryRole != "EndpointSlice"` then we needed to change `endpointSliceSupported = false` (if it had been true before). That was why I changed the type of `endpointSliceSupported` to a bool pointer.

But now rereading the code I realized we can check the value of `ServiceDiscoveryRole` inside `New()` in Prometheus/Prometheus Agent operator, right before calling `NewConfigGenerator()`. So no need to make `endpointSliceSupported` a bool pointer anymore.